src/aggregation/agg_result.rs

```diff
@@ -0,0 +1,37 @@
+use std::collections::HashMap;
+
+use crate::DocId;
+
+pub type AggregationResultTree = HashMap<String, AggregationResult>;
+
+pub enum AggregationResult {
+    BucketResult(BucketAggregationResult),
+    MetricResult(MetricResult),
+}
+
+pub enum MetricResult {
+    Average(i64),
+}
+
+pub struct BucketAggregationResult {
+    buckets: Vec<Box<BucketDataEntry>>,
```

Comment:
Why Box?

Reply:
I moved the Box now into BucketDataEntry to break up the circular structure

Reply:
I thought no Box was required. I mean it is already on the stack, and therefore should be sized.

Reply:
`BucketDataEntry` can have a sub aggregation tree